bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java

```diff
@@ -3655,7 +3655,7 @@ public boolean readAndDispatch () {
 		runDeferredEvents ();
 		return true;
 	}
-	return isDisposed () || runAsyncMessages (false);
+	return isDisposed () || runAsyncMessages (null); // handle multiple messages to avoid costly checks
```

Comment:
This is very sensitive area & should be tested on all 3 platforms.
I'm not sure a potential improvement for the synthetic use case of billions async messages is worth introducing a subtle regression.

Running "all" "asyncExec" pending tasks (even time bound) might "ignore" OS events triggered by one of them, so we will see OS native events handled "too late", and see strange effects no one will ever be able to trace to this change.